src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,48 +1,79 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  selectTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo, selectTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isInfoLoaded, setIsInfoLoaded] = useState(false);
+  const { completed, title, id } = selectedTodo;
+
+  const todoStatusStyle = completed ? 'success' : 'danger';
+  const todoStatus = completed ? 'Done' : 'Planned';
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        const userFromServer = await getUser(selectedTodo.userId);
+
+        setUser(userFromServer);
+      } catch {
+        setUser(null);
+      } finally {
+        setIsInfoLoaded(true);
```

Comment:
actually it is possible that you will have error in catch, than go to finally and above in code will think that you uploaded data, set error in catch case, and remove just remove loading in finally

Reply:
![guest](https://user-images.githubusercontent.com/89162624/220559077-fbf69afe-82d9-4ebd-b86e-e9bc96877353.png)
![loaded](https://user-images.githubusercontent.com/89162624/220559083-e6103fbd-3d83-4287-87cf-371b6033b899.png)
![loading](https://user-images.githubusercontent.com/89162624/220559085-e8c6bda0-8834-4843-a5b0-8f46df10706b.png)
I added different error messages for different cases